### PR TITLE
Improve archive logs naming

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: denis-tingajkin <denis.tingajkin@xored.com>

## Motivation and Context
1) Improve archive logs naming
2) Archive logs for examples
3) Fix bug with force delete
4) Fix bug with running on_fail scripts
5) Fix bug with cleaning namespaces for configs with special root
6) https://github.com/networkservicemesh/networkservicemesh/issues/1605
7) Add possible to store previous logs from pods

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [X] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by haiodo** [comment]
@denis-tingajkin Please create issues for failed testcase.

File: test/kubetest/log_utils.go [hunk 1]
```diff
@@ -109,7 +106,7 @@ func showPodLogs(k8s *K8s, t *testing.T, pod *v1.Pod) {
 					logrus.Errorf("Can't log in file, reason %v", logErr)
 					logTransaction(name, content)
 				} else {
-					logrus.Infof("Saved log for %v. Check dir %v", name, logsDir())
+					logrus.Infof("Saved log for %v. Check dir %v", name, logsDir()+name+".zip")
```
[the remark this answers is not in the document]
Comment (denis-tingajkin) [reply test/kubetest/log_utils.go]: thanks, fixed

File: .cloudtest.yaml [hunk 2]
```diff
@@ -26,11 +26,11 @@ executions:
     kubernetes-env:
       - KUBECONFIG
     on_fail: |
-      make k8s-delete-nsm-namespaces
+      cd ../.. && make k8s-delete-nsm-namespaces
```
[the remark this answers is not in the document]
Comment (denis-tingajkin) [reply .cloudtest.yaml]: used `cmd.Dir`

File: test/kubetest/log_utils.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func makeLogsSnapshot(k8s *K8s, t *testing.T) {
 }
 
 func archiveLogs(testName string) {
-	file, err := os.Create(filepath.Join(logsDir(), testName, DefaultArchiveFile))
+	file, err := os.Create(filepath.Join(logsDir(), testName+".zip"))
```
Comment (haiodo) [inline test/kubetest/log_utils.go]: @denis-tingajkin  Please talk with @artem-belov since we need a cluster name here as well.

**Comment by denis-tingajkin** [comment]
@artem-belov , @haiodo can we fix problem with interdomain in separate PR?

**Comment by haiodo** [comment]
Merged, filed  test is looked by @lobkovilya

**Pull request merged**